packages/wrangler/src/index.ts

```diff
@@ -783,18 +784,28 @@ export async function main(argv: string[]): Promise<void> {
 		}
 		throw e;
 	} finally {
-		// In the bootstrapper script `bin/wrangler.js`, we open an IPC channel, so
-		// IPC messages from this process are propagated through the bootstrapper.
-		// Make sure this channel is closed once it's no longer needed, so we can
-		// cleanly exit. Note, we don't want to disconnect if this file was imported
-		// in Jest, as that would stop communication with the test runner.
-		if (typeof jest === "undefined") process.disconnect?.();
-
-		// Close Sentry after disconnecting the IPC channel. Doing this before leads
-		// to hanging `workerd` processes.
-		// TODO(soon): work out why
-		await closeSentry();
+		try {
+			// In the bootstrapper script `bin/wrangler.js`, we open an IPC channel,
+			// so IPC messages from this process are propagated through the
+			// bootstrapper. Normally, Node's SIGINT handler would close this for us,
+			// but interactive dev mode enables raw mode on stdin which disables the
+			// built-in handler. Make sure this channel is closed once it's no longer
+			// needed, so we can cleanly exit. Note, we don't want to disconnect if
+			// this file was imported in Jest, as that would stop communication with
+			// the test runner.
+			if (typeof jest === "undefined") process.disconnect?.();
+
+			await closeSentry();
+		} catch (e) {
+			logger.error(e);
+			finallyError = e;
+		}
 	}
+	// If the first `try` block did NOT throw, but the `finally` did, throw that
+	// error. We have to do this outside the `finally` because if the first `try`
+	// DID `throw`, we want that to be rethrown from this function, not the error
+	// thrown by the `finally`.
+	if (finallyError !== undefined) throw finallyError;
```

Comment:
Alternative approach to consider...
Instead of `finallyError` variable, have a simple `error` variable that is assigned in the original catch block.
Then in the finally block try-catch, we can thrown the `e` if `error` is not defined.